src/PostgREST/DbStructure.hs

```diff
@@ -643,6 +662,46 @@ allM2ORels pgVer =
       else mempty) <>
     "ORDER BY conrelid, conname"
 
+allComputedRels :: Bool -> SQL.Statement () [Relationship]
+allComputedRels =
+  SQL.Statement sql HE.noParams (HD.rowList cRelRow)
+ where
+  sql = [q|
+    with
+    all_relations as (
+      select reltype
+      from pg_class
+      where relkind in ('v','r','m','f','p')
+    ),
+    computed_rels as (
+      select
+        p.pronamespace::regnamespace::text as schema,
+        p.proname as name,
+        split_part((p.proargtypes[0])::regtype::text, '.', 1) as rel_table_schema,
+        split_part((p.proargtypes[0])::regtype::text, '.', 2) as rel_table_name,
+        split_part(p.prorettype::regtype::text, '.', 1) as rel_ftable_schema,
+        split_part(p.prorettype::regtype::text, '.', 2) as rel_ftable_name,
```

Comment:
It looks like this will break when a table has a `.` in its name (probably it's the same with the schema name).

Reply:
Good catch. Fixed.